### PR TITLE
add lower bound slot condition on block gossip

### PR DESCRIPTION
**Pull request by djrtwo**
Add lower bound in beacon block gossip condition in response to Least Authority audit.

### CODE REVIEW

File: specs/phase0/p2p-interface.md [hunk 1]
```diff
@@ -251,6 +251,7 @@ There are two primary global topics used to propagate beacon blocks and aggregat
 
 - `beacon_block` - This topic is used solely for propagating new signed beacon blocks to all nodes on the networks. Signed blocks are sent in their entirety. The following validations MUST pass before forwarding the `signed_beacon_block` on the network
     - The block is not from a future slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance) -- i.e. validate that `signed_beacon_block.message.slot <= current_slot` (a client MAY queue future blocks for processing at the appropriate slot).
+    - The block is not from later than the latest finalized slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance) -- i.e. validate that `signed_beacon_block.message.slot > compute_start_slot_at_epoch(state.finalized_checkpoint.epoch)` (a client MAY choose to validate and store such blocks for additional purposes -- e.g. slashing detection, archive nodes, etc).
```
Comment (protolambda) [inline specs/phase0/p2p-interface.md]:
```suggestion
    - The block is not from before the latest finalized slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance) -- i.e. validate that `signed_beacon_block.message.slot > compute_start_slot_at_epoch(state.finalized_checkpoint.epoch)` (a client MAY choose to validate and store such blocks for additional purposes -- e.g. slashing detection, archive nodes, etc).
```
I read "not from later" as not after the event, while it should be the opposite. Reword for clarity, then good for merge :+1:

Comment (djrtwo) [reply specs/phase0/p2p-interface.md]: done!

Comment (djrtwo) [reply specs/phase0/p2p-interface.md]: FIXED THE TYPO SORRRY

File: specs/phase0/p2p-interface.md [hunk 2]
```diff
@@ -251,6 +251,7 @@ There are two primary global topics used to propagate beacon blocks and aggregat
 
 - `beacon_block` - This topic is used solely for propagating new signed beacon blocks to all nodes on the networks. Signed blocks are sent in their entirety. The following validations MUST pass before forwarding the `signed_beacon_block` on the network
     - The block is not from a future slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance) -- i.e. validate that `signed_beacon_block.message.slot <= current_slot` (a client MAY queue future blocks for processing at the appropriate slot).
+    - The block is from a slot greater than the latest finalized slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance) -- i.e. validate that `signed_beacon_block.message.slot > compute_start_slot_at_epoch(state.finalized_checkpoint.epoch)` (a client MAY choose to validate and store such blocks for additional purposes -- e.g. slashing detection, archive nodes, etc).
```
Comment (mkalinin) [inline specs/phase0/p2p-interface.md]: We should consider that depending on the configuration of the mesh a portion of such blocks may not reach slashing detectors and archive nodes at all.

Comment (djrtwo) [reply specs/phase0/p2p-interface.md]: Yeah, I'm a bit mixed on how these DoS vectors should be handled wrt slashing detection.

Let's chat about it on next networking call

**Pull request merged**